libomptarget/plugins/cuda/src/cupti.cpp

```diff
@@ -198,6 +196,8 @@ runtime_activities[] = {
   CUPTI_ACTIVITY_KIND_INVALID
 };
 
+static std::set<CUpti_ActivityKind> cupti_enabled_activities;
+static std::map<CUpti_ActivityKind, bool> cupti_paused_activities;
```

Comment:
What activities we have to pause.
If we use a set, update it during traversal is difficult.

Reply:
Can the activities being monitored or paused be different on different devices? It would be more general in cupti.cpp to have a set per device.

Reply:
I have a another idea:
We can enable/disable activities under different context. 
So let's change it to something like `map<CUcontext, map<CUpti_ActivityKind, bool> >`